### PR TITLE
Bug 2092337: os is empty in VM details page

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

Checking for guest agent OS data, if not exists fall back to check VM os annotation

### before:
os-missing

### after:
os-missing-after

VM is not running:
os-missing-after-2

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

**Comment by yaacov** [comment]
we should use the guest agent information:
https://github.com/kubevirt-ui/kubevirt-plugin/blob/main/src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabDetails/VirtualMachinesOverviewTabDetails.tsx#L52

**Comment by avivtur** [comment]
@yaacov ok I can use the guest agent data for os, should I fall back to any annotation from the VM?

File: src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx [hunk 1]
```diff
@@ -203,7 +205,15 @@ const VirtualMachineDetailsLeftGrid: React.FC<VirtualMachineDetailsLeftGridProps
           data-test-id={`${vm?.metadata?.name}-description`}
         />
         <VirtualMachineDescriptionItem
-          descriptionData={getOperatingSystemName(vm) || getOperatingSystem(vm)}
+          descriptionData={
+            guestAgentData?.os?.prettyName ||
+            guestAgentData?.os?.name ||
+            getOperatingSystemName(vm) ||
+            getOperatingSystem(vm) ||
+            vm?.spec?.template?.metadata?.annotations?.[ANNOTATIONS.os]
```
Comment (yaacov) [inline src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx:213]:
```suggestion
            (guestAgentData?.os?.prettyName || guestAgentData?.os?.name) ?? guestAgentIsRequired
```

Comment (yaacov) [reply src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx:213]: > @yaacov ok I can use the guest agent data for os, should I fall back to any annotation from the VM?

sorry missed that, answer above :-)

Comment (avivtur) [reply src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx]: updated, thanks

**Pull request merged**